src/Calculator.cpp

```diff
@@ -29,22 +31,22 @@ Calculator::Calculator(std::unique_ptr<IOperationFactory>&& factory)
 
 
 
-Result<double> Calculator::calculateResult(std::string equationString) const
+Result<double> Calculator::calculateResult(std::vector<std::string> infixVector) const
```

Comment:
MathExpression::getPopulatedEquation now does the same thing that Calculator::infixToPostfix used to do as far as separating the string into individual tokens (the loop in infixToPostfix used to be a for on infixString.length().) I changed it because initially I had MathExpression::getPopulatedEquation just drop the stored values for variables into the final equation string, but that led to issues where a negative variable being negated led to parse errors — eg: "-x" with x = -1 would result in "--1", leading to a missing operator error case (with the tokens being parsed as _ - -1 .) Now once Calculator gets an infix equation, every token is its own element in a vector, so it doesn't have to bother with parsing it, and MathExpression can handle if a value is negative by pulling the variable value, negating it, then putting whatever the result is into the vector.